tests/gireg.lua

```diff
@@ -36,8 +42,10 @@ function gireg.type_int8()
    checkv(R.test_int8(0), 0, 'number')
    checkv(R.test_int8(1), 1, 'number')
    checkv(R.test_int8(-1), -1, 'number')
-   checkv(R.test_int8(1.1), 1, 'number')
-   checkv(R.test_int8(-1.1), -1, 'number')
+   if not nativeIntegers then
```

Comment:
Would be nice to leave a comment here explaining why these are skipped. Or alternatively, since I expect that these will error out, it could be possible to add an `else` that checks via `pcall` for the error.

However, this raises the question on consistency: Some LGI versions now silently round while others error out. Should `check_integer` for Lua < 5.3 also check for integer values and throw an error? That way, behaviour would always be (more or less) the same.

Alternatively, the safer version would be to keep this rounding behaviour. That way the likelyhood of breaking someone's code is smaller.

Reply:
That's something I'm not 100% sure about. I think that throwing error on non-integer is good idea (that's what Lua itself does).

Whole change is done inside Lua core itself, so I'm don't really think that it's problem of lgi. Also, there are few places where < Lua 5.3 code actually use `lua_tonumber` on integer values.